exporter/datasetexporter/config.go

```diff
@@ -17,41 +17,84 @@ import (
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
 )
 
+const tracesExportSeparatorDefault = "."
+const tracesExportDistinguishingSuffix = "_"
+
 type TracesSettings struct {
+	// ExportSeparator is separator used when flattening exported attributes
+	// Default value: .
+	ExportSeparator string `mapstructure:"export_separator"`
+	// ExportDistinguishingSuffix is suffix used to be appended to the end of attribute name in case of collision
+	// Default value: _
+	ExportDistinguishingSuffix string `mapstructure:"export_distinguishing_suffix"`
```

Comment:
nit: move these to a separate struct that you embed on `TraceSettings` and `LogsSettings` to avoid duplication both on the struct definition and the default definition

Reply:
This was my initial thinking as well, that this should be single configuration shared by logs and traces. 

But then I have realised, that:
1.  the logic for logs and traces is completely different. For traces we are extracting just handful of properties from the resource and we completely ignore the scope. For logs people can extract everything.
2. Most people are extracting either logs or traces. So it's nice if you can define it in "your" section.
3. It was not obvious to me, what should be the name of the shared section - `export`, `extraction` - looks little bit strange to me.
4. Initially I was thinking, that none of these two things should be configurable. But if you are specifying prefixes, maybe you want to also specify the separator -> so I have made it configurable.

I do not think that any of those solution is significantly better. If it's fine for you, can we keep it as is - and we can observe, how people use it.

Reply:
I think it's fine if they are separate on the configuration exposed to the user because of the points you mentioned, I tend to be conservative when adding new options, but this is a personal preference and I don't want to impose that on you. It is also unclear to me what the name of the shared section should be if we make it a common section.

What I do think we can make is to avoid duplication at the Go level. We can define a struct:

```go
type exportSettings struct {
	// ExportSeparator is separator used when flattening exported attributes
	// Default value: .
	ExportSeparator string `mapstructure:"export_separator"`
	// ExportDistinguishingSuffix is suffix used to be appended to the end of attribute name in case of collision
	// Default value: _
	ExportDistinguishingSuffix string `mapstructure:"export_distinguishing_suffix"`
}
```

which we can then embed in the `LogsSettings` and `TraceSettings`:

```go
type TracesSettings struct {
    exportSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
    // existing fields go here
}


type LogsSettings struct {
    exportSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
    // existing fields go here
}
```

and have a common helper for defining the default:

```go
func newDefaultExportSettings() exportSettings {
    return exportSettings{
          ExportSeparator:            exportSeparatorDefault,
	  ExportDistinguishingSuffix: ExportDistinguishingSuffix,
     }
}
```

which we can use for the default settings:
```go
func newDefaultLogsSettings() LogsSettings {
	return LogsSettings{
		ExportResourceInfo:             logsExportResourceInfoDefault,
		ExportResourcePrefix:           logsExportResourcePrefixDefault,
		ExportScopeInfo:                logsExportScopeInfoDefault,
		ExportScopePrefix:              logsExportScopePrefixDefault,
		exportSettings: newDefaultExportSettings(),
		DecomposeComplexMessageField:   logsDecomposeComplexMessageFieldDefault,
		DecomposedComplexMessagePrefix: logsDecomposedComplexMessageFieldPrefixDefault,
	}
}
```

Does that make sense?

Reply:
Thanks! I haven't realised, that it's possible to do it like this. :)